publication-file/src/main/java/no/unit/nva/publication/file/upload/FileService.java

```diff
@@ -75,21 +79,24 @@ public InitiateMultipartUploadResult initiateMultipartUpload(SortableIdentifier
         return amazonS3.initiateMultipartUpload(request);
     }
 
-    public UploadedFile completeMultipartUpload(SortableIdentifier resourceIdentifier,
-                                                   CompleteUploadRequestBody completeUploadRequestBody,
-                                                   UserInstance userInstance) throws NotFoundException {
+    public File completeMultipartUpload(SortableIdentifier resourceIdentifier,
+                                        CompleteUploadRequest request, UserInstance userInstance)
+        throws NotFoundException, BadRequestException {
 
         var resource = fetchResource(resourceIdentifier);
 
-        var completeMultipartUploadRequest = completeUploadRequestBody.toCompleteMultipartUploadRequest(BUCKET_NAME);
+        var completeMultipartUploadRequest = request.toCompleteMultipartUploadRequest(BUCKET_NAME);
         var completeMultipartUploadResult = amazonS3.completeMultipartUpload(completeMultipartUploadRequest);
         var s3ObjectKey = completeMultipartUploadResult.getKey();
         var objectMetadata = getObjectMetadata(s3ObjectKey);
 
-        var file = constructUploadedFile(UUID.fromString(s3ObjectKey), objectMetadata, userInstance);
-
+        File file;
+        if (userInstance.isExternalClient() && request instanceof ExternalCompleteUploadRequest externalRequest) {
+            file = constructFileForExternalClient(UUID.fromString(s3ObjectKey), externalRequest, objectMetadata);
```

Comment:
We probably want to invent a new scope for our external clients here because isExternalClient() is not a very good discriminator whether you're allowed or not to directly publish stuff. We do have the file-upload scope today, but its not explicit enough I think. Current solution here is probably ok for now to get things working, but this is an issue we probably need to address soon because every time we do these non-specific access things, our code gets a bit more fuzzy and unreadable in my opinion at least :-)